docs/gallery/howto/autogen/if.py

```diff
@@ -0,0 +1,281 @@
+"""
+=====================================
+Flow control: Using ``if`` conditions
+=====================================
+"""
+
+# %%
+# Introduction
+# ============
+#
+# This tutorial provides a step-by-step guide on how to implement conditional ``if`` logic in WorkGraph using three
+# different methods:
+#
+# 1. **If context manager**
+# 2. **graph_builder decorator**
+# 3. **If Task**
+#
+# For simple cases, we recommend option 1), the ``If`` context manager approach, while option 2), the Graph
+# Builder provides additional advantages, see :doc:`graph_builder`.
+# Finally, option 3) uses the ``If`` ``Task`` directly without the context manager. This approach requires a lot of
+# boilerplate code and is generally not recommended.
+# It presents the node-graph programming approach to construct the ``If`` flow control element.
+#
+
+# %%
+# Setting up the AiiDA environment
+# --------------------------------
+#
+from aiida_workgraph import WorkGraph, task, If
+from aiida_workgraph.utils import generate_node_graph
+
+from aiida import load_profile
+
+load_profile()
+
+# %%
+# If conditional workflow
+# =======================
+#
+# Suppose you have the following Python workflow:
+
+
+def add(x, y):
+    return x + y
+
+
+def multiply(x, y):
+    return x * y
+
+
+# First addition
+result = add(1, 1)
+
+# Conditionally execute addition or multiplication
+if result < 0:
+    result = add(result, 2)
+else:
+    result = multiply(result, 2)
+
+# Last addition
+result = add(result, 1)
+
+print("Result is", result)
+
+# %%
+# Now, to convert this workflow into a WorkGraph, we first convert the Python functions, ``add`` and ``multiply`` to
+# tasks:
+
+
+@task.calcfunction()
+def add(x, y):
+    return x + y
+
+
+@task.calcfunction()
+def multiply(x, y):
+    return x * y
+
+
+# %%
+# Using the ``If`` context manager
+# --------------------------------
+#
+# WorkGraph provides the ``If`` context manager that allows you to define conditional logic in your workflow. The ``If``
+# block encapsulates all its child tasks, which are executed based on the defined conditions.
+
+# %%
+# To then define define the conditional logic for the workflow, we use the ``If`` context manager provided by
+# AiiDA WorkGraph.
+# Note that to use the ``If`` context manager, we also need to construct our top-level WorkGraph via the context manager
+# approach. More information on this is provided in :doc:`../concept/workgraph`.
+
+with WorkGraph("if_context") as wg:
+    result = add(x=1, y=1)
+    with If(result < 0):
+        r1 = add(x=result, y=2)
+        wg.ctx.result = r1
+    with If(result >= 0):
+        r2 = multiply(x=result, y=2)
+        wg.ctx.result = r2
+
+    # -----------------------------------------
+    result = add(x=wg.ctx.result, y=1)
```

Comment:
Due to the change in this [PR](https://github.com/aiidateam/aiida-workgraph/pull/529), we always return the `task.outputs`, in order to get `task.outputs.result`, we need do it manually.
```suggestion
with WorkGraph("if_context") as wg:
    result = add(x=1, y=1).result
    with If(result < 0):
        r1 = add(x=result, y=2).result
        wg.ctx.result = r1
    with If(result >= 0):
        r2 = multiply(x=result, y=2).result
        wg.ctx.result = r2

    # -----------------------------------------
    result = add(x=wg.ctx.result, y=1).result
```

Reply:
OK, nice, thanks! Commited